### ISSUE TITLE
Tabmenu: Loading global styles after page is loaded looks weird

### ISSUE THREAD

**Issue by gslopez** [primefaces/primereact#5188]
### Describe the bug

I have this super simple code
```
import React, {useState} from 'react';
import {TabMenu} from 'primereact/tabmenu';
import {MenuItem} from 'primereact/menuitem';


type NavbarProps = {
    onPageSelected?: (page: string) => void,
    menuItems?: MenuItem[],
}

export const Navbar = ({menuItems, onPageSelected}: NavbarProps) => {
    const [activeIndex, setActiveIndex] = useState(0)

    return (
        <TabMenu
            activeIndex={activeIndex}
            model={
                menuItems
            }
            onTabChange={(e) => {
                setActiveIndex(e.index);
                onPageSelected?.(e.value.id as string)
            }}
        />
    )
}

```

When the page first loads, it looks like this first for like 0.5 seconds

<img width="463" alt="Screenshot 2023-10-29 at 3 24 56 PM" src="https://github.com/primefaces/primereact/assets/2165906/3b20fa7b-ee12-475e-b434-82165ab3e36f">

and then it changes to how it should really look

<img width="685" alt="Screenshot 2023-10-29 at 3 24 47 PM" src="https://github.com/primefaces/primereact/assets/2165906/20ea7343-52d6-42cf-bac2-5c2054040778">

After some debugging I found out it is because when the component is loaded, it loads this global style 
<img width="545" alt="Screenshot 2023-10-29 at 3 27 42 PM" src="https://github.com/primefaces/primereact/assets/2165906/271b9872-e4bc-4c94-a533-d8d267375432">

https://github.com/primefaces/primereact/blob/master/components/lib/tabmenu/TabMenuBase.js#L79
https://github.com/primefaces/primereact/blob/master/components/lib/componentbase/ComponentBase.js#L664

I recorded this video with the issue on the codesandbox i created

https://github.com/primefaces/primereact/assets/2165906/01eb40c2-e062-40ee-8934-3355b858f502

I would expect either
1. The component should be auto-sufficient (i.e it looks properly on first load)
2. To have a way to import the components css globally directly, next to the theme and min imports (something like `import "primereact/components/tabmenu.css`

### Reproducer

https://codesandbox.io/s/primereact-test-forked-xsr6gj?file=/src/index.js

### PrimeReact version

10.0.7

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

**Comment by melloware** [comment]
I agree I have seen this same flash flicker when it renders.

**Comment by melloware** [comment]
@gslopez check this out can you try this: https://codesandbox.io/s/primereact-test-forked-9cl6y8?file=/src/styles.css

I merged all of the base styles into a single CSS file to load.  If this meets your needs I think I can propose something clever.

**Comment by melloware** [comment]
Updated with more complete version of the final CSS: https://codesandbox.io/s/primereact-test-forked-9cl6y8?file=/src/styles.css

**Comment by gslopez** [comment]
hey @melloware! Thanks for the quick answer. Yeah, if I add `import "./styles.css";` to that codesandbox it seems to be working fine. I think this is good enough for my current project 🙏

**Comment by alaordev** [comment]
@melloware any date prediction of 10.0.10 release? Facing the same issue here

**Comment by melloware** [comment]
NextJS 14 reproducer: https://stackblitz.com/edit/stackblitz-starters-mqapwn?file=app%2Fglobals.css,app%2Flayout.tsx

**Comment by axlerk** [comment]
Facing the same issue while migrating to a major version 8 → 10 on my project. Any news here? It's look like a blocker for me

**Comment by melloware** [comment]
You can use my workaround posted above

**Comment by lsegal** [comment]
> You can use my workaround posted above

@melloware do you mean this?

> Updated with more complete version of the final CSS: https://codesandbox.io/s/primereact-test-forked-9cl6y8?file=/src/styles.css

This codesandbox still shows a FOUC on first load. It's a little less noticeable but it still occurs. A little difficult to show with Chrome Inspector (it limits how big the Visual Preview window can be) but the following shows the list elements along with the final rendered content 1ms later:

![image](https://github.com/primefaces/primereact/assets/686/5ad9beb6-0378-4991-9cb5-eb116b2bdb32)
![image](https://github.com/primefaces/primereact/assets/686/0f40e08a-88e4-4f9d-979e-64b5664d53f2)

**Comment by melloware** [comment]
Yes it doesn't eliminate it it just makes it less jarring by preloading CSS.

**Comment by axlerk** [comment]
Is it possible to somehow escalate this issue? Add more likes or idk :) Looks important.

**Comment by melloware** [comment]
@axlerk PrimeTek is aware of this issue its on their radar.

**Comment by emirhancopoglu** [comment]
any news?

**Comment by thingersoft** [comment]
> any news?

+1

**Comment by cdllc** [comment]
+another

**Comment by emirhancopoglu** [comment]
+++

**Comment by starlocater** [comment]
++++

**Comment by axlerk** [comment]
@melloware could you update the workaround for version 10.6.6 please? Until we have a solution

**Comment by melloware** [comment]
@axlerk i assume you means some styles are missing from my giant CSS file?

**Comment by acho1833** [comment]
It’s not working on my end with latest version.  I think styles are loaded fine but it uses fallback font sans-serif before it switches to ‘Inter var’.  Is there a way to block until font is loaded? I did try font-display: block but no go.

**Comment by acho1833** [comment]
Thanks for the solution above.  It partially made it better but there are some components that still do fouc like dropdown component.  For now, almost guranteed way of rendering is to give about 500ms before rendering the component.  I really don't like it but I can't think of any work around solution this.  Anyone with better idea?

**Comment by acho1833** [comment]
Now I get what melloware is saying now.  PrimeReact dynamically adds the style but after the 'component' have been rendered.  For future people looking into this, see the screen shot.  The left you'll notice 

Now I get what @melloware is saying now, PrimeReact adds the styles 'after' the component have already been rendered.
See the left side of the screenshot and you'll notice that 'base', 'global', 'splitbutton', are being added dynamically.  On the right side of the screen, you don't and that's why it's showing FOUC.

![image](https://github.com/primefaces/primereact/assets/170477364/59c28cec-6d1d-4210-9c6d-071b56bb96fb)

I think I can come with decent 'workaround' solution without putting those css somewhere.  My idea is the following.  Please let me know if my logic can be improved or flawed.

1.  Make the <body> to be display:none.  Something like
```
<body style={{display: "none"}}>
```
2.  Add a <script> tag, that checks for added styles with the attribute 'data-primereact-style-id'.  Check that at least 3 of them exists (Imagine those are 'base', 'global', 'common') and maybe add another 50ms and then change the body style back to 'block'/'flex'/or whatever...

Hopefully this will work as temporary workaround...  
```

that in the root layout, I'll add <script> tag that will check if style have been applied (maybe at least at least 3 style with attribute 'data-primereact-style-id] exists and then

**Comment by melloware** [comment]
@acho1833 yes you understand the issue now.  PrimeTek I think is working on a long term solution after their next big refactor.

**Comment by acho1833** [comment]
My theory seems to work and I think it's good to go now.  To me, this is enough until the fix and looks perfect to me.
`
export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <head>
      <style
          dangerouslySetInnerHTML={{
            __html: `
              body {
                display: none;
              }
            `,
          }}
        />
        <script
          dangerouslySetInnerHTML={{
            __html: `
                document.addEventListener('DOMContentLoaded', function() {
                    const intervalId = setInterval(() => {
                        const elements = document.querySelectorAll('[data-primereact-style-id]');
                        if (elements.length >= 3) {
                            document.body.style.display = 'block';
                            clearInterval(intervalId);
                        }
                    }, 50);
                });
            `,
          }}
        />
      </head>
      <body className={inter.className} >
        <PrimeReactProvider>{children}</PrimeReactProvider>
      </body>
    </html>
  );
}
`

@melloware looking forward for next release.  Is that real soon? know any rough ETA?

**Comment by melloware** [comment]
I don't work for PrimeTek so I don't know the internal details of their schedule or roadmap.

**Comment by Saul-Sosa-Diaz** [comment]
I have found a solution for nextjs in another [issue](https://github.com/vercel/next.js/issues/62940) that directs to this [nextjs page](https://nextjs.org/docs/messages/react-hydration-error#solution-1-using-useeffect-to-run-on-the-client-only), at least this  worked for me.

**Comment by thingersoft** [comment]
I see that v11 is marked as the milestone for resolving this issue, but I wanted to point out that the online v11 alpha demo still shows a noticeable FOUC.  
For example, try loading or refreshing the `rating` component page: https://v11.primereact.org/docs/components/rating
